host-components/src/main.rs

```diff
@@ -0,0 +1,303 @@
+        let mut kllvec = new_empty_KeepLoaderList();
+        for existing_keeps in glob("/tmp/enarx-keep-*.sock").expect("Failed to read glob pattern") {
+            //println!("keep-loader = {:?}", &existing_keeps);
+            match existing_keeps {
+                Ok(path) => {
+                    let stream_result = UnixStream::connect(path.clone());
+                    match stream_result {
+                        Ok(mut stream) => {
+                            println!("Able to connect to {:?}", path.display());
+                            //TODO - add to list
+                            //this is what we'll add, but first we need to contact the
+                            //keep-loader and find out what the kuuid and app_loader_bind_port are
+                            let jsoncommand = JsonCommand {
+                                commandtype: String::from(KEEP_INFO_COMMAND),
+                                commandcontents: "".to_string(),
+                            };
+                            let serializedjson = serde_json::to_string(&jsoncommand)
+                                .expect("problem serializing data");
+
+                            //println!("Sending JSON data\n{}", serializedjson);
+                            &stream
+                                .write_all(&serializedjson.as_bytes())
+                                .expect("failed to write");
+                            //now get a reply
+                            let deserializer = serde_json::Deserializer::from_reader(stream);
+                            let iterator = deserializer.into_iter::<serde_json::Value>();
+                            for json_pair in iterator {
+                                println!(
+                                    "Got a reply from {}, which is {:?}",
+                                    path.display(),
+                                    &json_pair
+                                );
+                                match json_pair {
+                                    Ok(value) => {
+                                        let keeploader: KeepLoader =
+                                            serde_json::from_value(value).unwrap();
+                                        let mut keeploaderlist = kllvec.lock().await;
+                                        println!(
+                                            "keeploader on {} has kuuid {}",
+                                            path.display(),
+                                            keeploader.kuuid,
+                                        );
+                                        keeploaderlist.push(keeploader);
+                                        println!("Pushed keeploader to list");
+                                    }
+                                    Err(e) => println!("not a useful reply"),
+                                }
+
+                                break;
```

Comment:
It might be easier to read, if this is written as `iterator.next()`, rather than using a big for loop ending with a `break`?

Reply:
I'm not sure we're going to end up using this code - it's there almost as a placeholder (and not used for now), so I'm not sure it's worth spending too much time improving it.

Reply:
I've marked this as a "TODO" - it currently works, but I agree that it could be improved.